mlir/include/mlir/Dialect/LLVMIR/NVVMOps.td

```diff
@@ -819,6 +819,26 @@ def NVVM_VoteBallotOp :
   let hasCustomAssemblyFormat = 1;
 }
 
+def NVVM_VoteAllSyncOp : NVVM_Op<"vote.all.sync">,
```

Comment:
I think we can use the existing op `nvvm.vote.ballot.sync` and extend its mode. so it's close to ptx.

Reply:
Ok. I can update it this way.